vigor/models.py

```diff
@@ -1,11 +1,129 @@
+    gender = models.CharField(max_length = 1, choices = GENDERS, null = True, blank = True)
+    birth_year = models.IntegerField(null = True, blank = True)
+    birth_month = models.IntegerField(null = True, blank = True)
+    birth_day = models.IntegerField(null = True, blank = True)
+    ethnicity = models.CharField(max_length = 25, choices = ETHNICITIES, default = 'White')
+    height_feet = models.IntegerField(null = True, blank = True)
+    height_inches = models.IntegerField(null = True, blank = True)
+    weight = models.IntegerField(null = True, blank = True)
+    determination = models.CharField(max_length = 15, choices = DETERMINATIONS, default = 'Gain Weight')
+    time_zone = TimeZoneField(null = True, blank = True)
+    date_format = models.CharField(max_length=10, choices=DATE_FORMATS, default='MM/DD/YYYY')
+
+    def get_today(self, pobj = False):
+        today = datetime.datetime.utcnow().replace(tzinfo = utc)
+        if self.time_zone:
+            today = today.astimezone(self.time_zone)
+            if pobj:
+                return today
+        return today.strftime(self.get_date_format())
+
+    def get_date_format(self):
+        if self.date_format == 'MM/DD/YYYY':
+            return '%m/%d/%Y'
+        elif self.date_format == 'DD/MM/YYYY':
+            return '%d/%m/%Y'
+
+    def is_male(self):
+        return self.gender == 'M'
+
+    def is_female(self):
+        return self.gender == 'F'
+
+    def get_age(self):
+        if self.birth_year and self.birth_month and self.birth_day:
+            today = self.get_today(pobj = True)
+            return today.year - self.birth_year - ((today.month, today.day) < (self.birth_month, self.birth_day))
+
+    def is_adult(self):
+        return self.get_age() > 18
+
+    def get_calorie_regimen(self):
+        if self.determination == 'Gain Weight':
+            if self.gender == 'M':
+                return 10 * self.weight + 6.25 * (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() + 505
+            elif self.gender == 'F':
+                return 10 * self.weight + 6.25 (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() + 500 - 161
+        elif self.determination == 'Maintain Weight':
+            if self.gender == 'M':
+                return 10 * self.weight + 6.25 * (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() + 5
+            elif self.gender == 'F':
+                return 10 * self.weight + 6.25 (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() - 161
+        elif self.determination == 'Lose Weight':
+            if self.gender == 'M':
+                return 10 * self.weight + 6.25 * (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() + 5 - 500
+            elif self.gender == 'F':
+                return 10 * self.weight + 6.25 (((self.height_feet * 12) + self.height_inches) * 2.54) - 5 * self.get_age() - 161 - 500
```

Comment:
I've pushed an update to simplify this equation, and to reduce hardcoded numbers, by assigning them to vars. The other numbers, I've left because they are a collection of other numbers precalculated. Is this okay still?